### PR TITLE
fix: move rebaseWhen=never check earlier

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Refactors rebaseWhen=never check to earlier to prevent unnecessary branch updates.

## Context:

Performance

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/branch/index.ts [hunk 1]
```diff
@@ -345,6 +345,13 @@ export async function processBranch(
     ) {
       logger.debug('Manual rebase requested via Dependency Dashboard');
       config.reuseExistingBranch = false;
+    } else if (branchExists && config.rebaseWhen === 'never') {
```
Comment (viceice) [inline lib/workers/branch/index.ts:348]: So you are no longer able to trigger a rebase by using the checkbox?

Comment (rarkins) [reply lib/workers/branch/index.ts:348]: Shouldn't it still work due to the `if` above?

Comment (viceice) [reply lib/workers/branch/index.ts:348]: `config.rebaseRequested` isn't checked there. 🤔

Comment (rarkins) [reply lib/workers/branch/index.ts:348]: I'll check this manually

**Comment by rarkins** [comment]
I verified that with the latest commit you can request a rebase within a PR and it works

**Review by viceice — APPROVED** [review]

**Pull request merged**